### ISSUE TITLE
check on MG26x bias reweight using  GEN level events 

### ISSUE THREAD

**Issue by menglu21** [cms-sw/genproductions#1482]
Hi, @bendavid @kdlong i have got some results on the mg26x bias reweight, the process is 
generate p p > ell+ ell- j [QCD] @0 for both no_bias and with_bias. 
while the pythia setting is 
processParameters = cms.vstring('JetMatching:setMad = off',
            'JetMatching:scheme = 1',
            'JetMatching:merge = on',
            'JetMatching:jetAlgorithm = 2',
            'JetMatching:etaJetMax = 999.',
            'JetMatching:coneRadius = 1.',
            'JetMatching:slowJetPower = 1',
            'JetMatching:qCut = 30.',
            'JetMatching:doFxFx = on',
            'JetMatching:qCutME = 10.',
            'JetMatching:nQmatch = 5',
            'JetMatching:nJetMax = 2',
            'TimeShower:mMaxGamma = 4.0')

 the bias function is "bias_wgt = (max_ptj/1000)**3".

more detail in the attachmet.

![screenshot from 2017-11-03 20-46-48](https://user-images.githubusercontent.com/19426117/32393004-2e5d0aec-c0d8-11e7-9144-f50ab23c5df5.png)
![ptjet_log](https://user-images.githubusercontent.com/19426117/32393289-35681ace-c0d9-11e7-9d9a-d44249ae5179.png)

    
[dyellell1j_5f_NLO_FXFX_26_simple_bias_run_card.txt](https://github.com/cms-sw/genproductions/files/1442300/dyellell1j_5f_NLO_FXFX_26_simple_bias_run_card.txt)
[dyellell1j_5f_NLO_FXFX_26_simple_bias_proc_card.txt](https://github.com/cms-sw/genproductions/files/1442301/dyellell1j_5f_NLO_FXFX_26_simple_bias_proc_card.txt)
[dyellell1j_5f_NLO_FXFX_26_simple_bias_cutsf.txt](https://github.com/cms-sw/genproductions/files/1442302/dyellell1j_5f_NLO_FXFX_26_simple_bias_cutsf.txt)
[SMP-RunIISummer15GS-00625_1_cfgpy.txt](https://github.com/cms-sw/genproductions/files/1442306/SMP-RunIISummer15GS-00625_1_cfgpy.txt)

**Comment by kdlong** [comment]
Thanks Meng, this looks quite nice! Can you give a little information about how the plots/data are obtained?

-What objects are you using in the plots? ls it LHE-level info accessed from the LHE file or from the LHE EDM file?
- Do you plan to make the sample plots using the full definition of the process we use in CMS (merged with 2 NLO jets) using GEN information rather than ? I think it would be best to do this with a GEN-only sample.

**Comment by perrozzi** [comment]
Hi, adding Adinda (@adewit) who is also looking at this

**Comment by menglu21** [comment]
Hi Kenneth, 
* i did not post the lhe plot here. the plots are from gen-level, and they are the same but the bottom one is in log-scale. 
* yes, i plan to proceed on with the inclusive DY sample to get some plots.

**Comment by adewit** [comment]
Hi, I'm a bit late to the party but I have been looking at this for DY with up to 1 jet, trying to enhance the Z pT tail, some slides are attached. Here everything seems to be working too, although care has to be taken to make sure there is a reasonable constant term so a sensible number of 0 jet events are generated, so as not to end up with huge uncertainties.
[BiasWeightValidation_0811_AdeWit.pdf](https://github.com/cms-sw/genproductions/files/1453917/BiasWeightValidation_0811_AdeWit.pdf)

**Comment by perrozzi** [comment]
Thanks a lot Meng and Adinda!

**Comment by bendavid** [comment]
Hi, as a final sanity check before declaring this technically validated, it would also be useful to check this with the inclusive W process, starting from https://github.com/cms-sw/genproductions/tree/mg26x/bin/MadGraph5_aMCatNLO/cards/examples/wellnu012j_5f_NLO_FXFX

The definition of the W pt of course would need to be modified, adapting from the pt-binned cards
https://github.com/cms-sw/genproductions/blob/pre2017/bin/MadGraph5_aMCatNLO/cards/production/13TeV/WJetsToLNu_Wpt_012j_5f_NLO_FXFX/WJetsToLNu_012j_Wpt-100To250_5f_NLO_FXFX/WJetsToLNu_012j_Wpt-100To250_5f_NLO_FXFX_cuts.f#L82-L85

**Comment by bendavid** [comment]
It would also be good to explicitly check the jet multiplicity both at LHE and genjet level.

At LHE level this can be done both by just counting the number of outgoing partons, but also by plotting the value of the "npNLO" tag.  In fact both would be useful, since the latter corresponds to the actual matrix element/process used.

**Comment by menglu21** [comment]
hi all, i have done the validation of process DY+1jet, i just attach all the file i got as i do not know how to present it in a better way. and pleasae go to https://indico.cern.ch/event/679779/contributions/2784192/attachments/1555941/2446789/Validation_of_MG_26x.pdf  to see some production detail.

[aa.tar.gz](https://github.com/cms-sw/genproductions/files/1457919/aa.tar.gz)

**Comment by adewit** [comment]
Hi, I've now added some checks on the number of partons, npNLO, and number of gen jets, as well as the W+ up to 1 jet process. The checks are summarised in the attached slides, from slide 10 onwards (the first 9 slides come from the slides I posted here previously). In summary: the number of gen jets/partons are compatible between biased and unbiased productions. In the case of W+up to 1 jet the discrepancies between unbiased and biased distributions seem to be a little bit larger than in the case of DY+up to 1 jet, although still mostly compatible.
[BiasWeightValidation_1311_AdeWit.pdf](https://github.com/cms-sw/genproductions/files/1466464/BiasWeightValidation_1311_AdeWit.pdf)
